### PR TITLE
proj: fix allow cmake helpers install

### PR DESCRIPTION
**Pull request by leolchat**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
Not fully there is a warning from before, about the description having the name of the formula in it.
-----

When using the configure, make route, proj4 doesn't install the cmake helper scripts, while using cmake to build it does.

I chose to set it as an option since it adds a dependency on cmake, but it might be that most people are willing to have the cmake helpers installed (a quick google for FindProj shows that many are reimplementing it because it is not installed by default).

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
I reckon because CMake is a build-time-only dependency that many other things depend on it's worth making this on unconditionally.

**Comment by leolchat** [comment]
You mean no option just switch the build to using cmake ? I can certainly do that.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Switching to cmake seems to change the library number, which is a bug that needs to be reported upstream and fixed before we can merge this, unless we revision bump all the reverse dependencies to accommodate the different library number.

**Comment by fxcoudert** [comment]
Upstream's bug report is there: https://github.com/OSGeo/proj.4/issues/378
Nothing moving so far, so let's close this PR for now and wait until upstream gets it fixed!

**Pull request closed without merging**
